### ISSUE TITLE
Tutorial 3 from Code Examples

### ISSUE THREAD

**Issue by yufan-zhang** [NetLogo/NetLogo#1714]
I didn't write my own code. I am using Tutorial 3 as is.

I have observed the same issue while trying to reproduce the code from the tutorial, but I have included the parameters below for reference. (Make the ticking slow to observe.)
- number: 1
- show-energy?: on
- birth-energy: 30
- energy-from-grass: 5

The moment when the turtle has more energy than birth-energy, it hatches, but its own energy does not get reduced by the birth-energy (or at all), nor does the new-born take on birth-energy. I don't think that's the intended behavior.

I played around, and found out that in the reproduce function, once past the if-condition, only the first line gets executed, i.e. either the energy gets reduced or hatching happens, but not both.

I just finished the tutorials today, so forgive me if I did something stupid. But I have double-checked.

Thanks.

**Comment by charlesres** [comment]
Hi, @yufan-zhang, do you mind providing which version of NetLogo you are using with the model. With the issue related to Tutorial 3, I would recommend sending an email, linking this issue, to ccl-models@ccl.northwestern.edu, since discussions on behaviors can be covered in more detail in the mailing list.

**Comment by charlesres** [comment]
@yufan-zhang. After running the model on NetLogo 6.0.4, I see that the parent turtle loses its energy and produces an offspring, which contains the birth-energy of 30. The reproduce function from my version looks like this:
```
to reproduce
  ask turtles [
    if energy > birth-energy [
    set energy energy - birth-energy  ;; take away birth-energy to give birth
    hatch 1 [ set energy birth-energy ] ;; give this birth-energy to the offspring
    ]
  ]
end
```
Is this what you are seeing?

**Comment by yufan-zhang** [comment]
Sorry I closed and opened and closed the issue. This is my first time using Github to report and I didn't know what I was doing.

I found out what my problem was: I was confused between the parent and the child, because as soon as a child is born, it takes on enough energy to hatch, so I thought the child is the parent, and parent (who has lost almost all energy) as the child.

Yes I see the same code that you posted. Once I changed the initial energy of newborns to a smaller constant (say 5), things became clear.

Thank you very much for your support.
